lib/sidekiq-scheduler/scheduler.rb

```diff
@@ -267,10 +267,8 @@ def new_job(name, interval_type, config, schedule, options)
         if job_enabled?(name)
           conf = SidekiqScheduler::Utils.sanitize_job_config(config)
 
-          # `SidekiqScheduler::Utils.calc_cron_run_time` only works with a job running periodically.
-          # Also, `SidekiqScheduler::Utils.calc_cron_run_time` is only required jobs are execute frequently. So only call it if really needed.
-          if job.is_a?(Rufus::Scheduler::CronJob) && job.cron_line.rough_frequency <= 60
-            idempotent_job_enqueue(name, SidekiqScheduler::Utils.calc_cron_run_time(job.cron_line, time.utc), conf)
+          if job.is_a?(Rufus::Scheduler::CronJob)
+            idempotent_job_enqueue(name, SidekiqScheduler::Utils.calc_cron_run_time(job.cron_line, time.to_t), conf)
```

Comment:
In my personal opinion, passing an instance of `EtOrbi::EoTime` as is better.

* `EtOrbi::EoTime` only holds a seconds. So no need to call `round`.
* `EtOrbi::EoTime` has `#-` and it allow to specify an instance of `EtOrbi::EoTime` . So we can get `next_t` and previous_t` without converting to `Time` instance. This allows us to reduce objects.
  * https://github.com/floraison/et-orbi/blob/ac15d51140720db6b82fa8e84b10d2f36f41ec24/lib/et-orbi/time.rb#L252

Reply:
So `EtOrbi::EoTime` holds the seconds as a float and Rufus is using `EoTime.now` which uses `Time.now.to_f` which contains sub-seconds. Removing the sub-seconds was something I discovered after seeing multiple jobs get scheduled twice consistently 1 second after the scheduled time. So I think the removal of sub-seconds is important.

I do think passing `EtOrbi::EoTime` is better but the codebase is marked as expecting a `Time` object everywhere and uses `Time.now` as defaults. I think maybe just converting everything to `Time` in the gem is better to stay consistent. As far as I can tell `EtOrbi::EoTime` is only in this codebase in `SidekiqScheduler::Scheduler#new_job` block and `SidekiqScheduler::Utils.new_rufus_scheduler` in the `on_post_trigger` callback.